x-pack/plugins/apm/public/components/app/correlations/correlations_chart.tsx

```diff
@@ -149,93 +155,172 @@ export function CorrelationsChart({
-          }
-        />
-        <AreaSeries
-          id={i18n.translate(
-            'xpack.apm.correlations.latency.chart.overallLatencyDistributionLabel',
-            { defaultMessage: 'Overall latency distribution' }
+        <Chart>
+          <Settings
+            rotation={0}
+            theme={chartTheme}
+            showLegend
+            legendPosition={Position.Bottom}
+            onBrushEnd={onChartSelection}
+          />
+          {selectionAnnotation !== undefined && (
+            <RectAnnotation
+              dataValues={selectionAnnotation}
+              id="rect_annotation_1"
+              style={{
+                strokeWidth: 1,
+                stroke: '#e5e5e5',
+                fill: '#e5e5e5',
+                opacity: 0.9,
+              }}
+              hideTooltips={true}
+            />
           )}
-          xScaleType={ScaleType.Log}
-          yScaleType={ScaleType.Log}
-          data={overallHistogram}
-          curve={CurveType.CURVE_STEP_AFTER}
-          xAccessor="key"
-          yAccessors={['doc_count']}
-          color={euiTheme.eui.euiColorVis1}
-          fit="lookahead"
-        />
-        {Array.isArray(histogram) &&
-          field !== undefined &&
-          value !== undefined && (
-            <AreaSeries
-              id={i18n.translate(
-                'xpack.apm.correlations.latency.chart.selectedTermLatencyDistributionLabel',
+          {typeof markerCurrentTransaction === 'number' && (
+            <LineAnnotation
+              id="annotation_current_transaction"
+              domainType={AnnotationDomainType.XDomain}
+              dataValues={[
+                {
+                  dataValue: markerCurrentTransaction,
+                  details: i18n.translate(
+                    'xpack.apm.correlations.latency.chart.currentTransactionMarkerLabel',
+                    {
+                      defaultMessage: 'Current sample',
+                    }
+                  ),
+                },
+              ]}
+              style={getAnnotationsStyle(euiPaletteColorBlind()[0])}
+              marker={i18n.translate(
+                'xpack.apm.correlations.latency.chart.currentTransactionMarkerLabel',
                 {
-                  defaultMessage: '{fieldName}:{fieldValue}',
-                  values: {
-                    fieldName: field,
-                    fieldValue: value,
-                  },
+                  defaultMessage: 'Current sample',
                 }
               )}
-              xScaleType={ScaleType.Log}
-              yScaleType={ScaleType.Log}
-              data={histogram}
-              curve={CurveType.CURVE_STEP_AFTER}
-              xAccessor="key"
-              yAccessors={['doc_count']}
-              color={euiTheme.eui.euiColorVis2}
+              markerPosition={'top'}
             />
           )}
-      </Chart>
-      <EuiSpacer size="s" />
+          <LineAnnotation
+            id="annotation_1"
```

Comment:
nit: I think it's better to rename this id to something more relevant

Reply:
Tweaked in 13b91a059bc.